src/App.tsx

```diff
@@ -17,18 +54,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filterOption={selectedFilterOption}
+                setFilter={setSelectedFilterOption}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!todosAreLoaded && <Loader />}
+              <TodoList
+                todos={filteredTodos}
+                showModal={setShowModal}
+                selectedTodo={selectedTodo}
+                selectTodo={setSelectedTodo}
+              />
```

Comment:
Probably would be better to show only one of them in the same time 
```suggestion
              {!todosAreLoaded 
                ? <Loader />
                : <TodoList
                  todos={filteredTodos}
                  showModal={setShowModal}
                  selectedTodo={selectedTodo}
                  selectTodo={setSelectedTodo}
                />
              }
              
```

Reply:
I added the condition, thank you!